python/mxnet/profiler.py

```diff
@@ -20,28 +20,34 @@
 # pylint: disable=too-many-branches, too-many-statements
 """Profiler setting methods."""
 from __future__ import absolute_import
-
 import ctypes
-from .base import _LIB, check_call, c_str
+from .base import _LIB, check_call, c_str, ProfileHandle, c_str_array
 
-def profiler_set_config(mode='symbolic', filename='profile.json'):
+def profiler_set_config(flags):
     """Set up the configure of profiler.
 
     Parameters
     ----------
-    mode : string, optional
-        Indicates whether to enable the profiler, can
-        be 'symbolic', or 'all'. Defaults to `symbolic`.
-    filename : string, optional
-        The name of output trace file. Defaults to 'profile.json'.
+    flags : list of key/value pair tuples
+        Indicates configuration parameters
+          profile_all : boolean, all profile types enabled
```

Comment:
mode='all|symbolic|imperative|api|memory' to keep backward compatibility?

Reply:
...